pkg/config/config.go

```diff
@@ -124,30 +132,48 @@ func CheckRegistryConfiguration(r Registry) error {
 		return fmt.Errorf("a registry requires a type")
 	}
 
-	errorWithType := func(info string) error {
-		return fmt.Errorf(`registry of type "%s" %s`, r.Type, info)
-	}
-
 	registry, _ := types.ParseRegistry(r.Type)
 	switch registry {
 	case types.RegistryAWS:
-		if r.AWS.Region == "" {
-			return errorWithType(`requires a field "region"`)
-		}
-		if r.AWS.AccountID == "" {
-			return errorWithType(`requires a field "accountdId"`)
-		}
+		return validateAWSRegistry(r)
 	case types.RegistryGCP:
-		if r.GCP.Location == "" {
-			return errorWithType(`requires a field "location"`)
-		}
-		if r.GCP.ProjectID == "" {
-			return errorWithType(`requires a field "projectId"`)
-		}
-		if r.GCP.RepositoryID == "" {
-			return errorWithType(`requires a field "repositoryId"`)
-		}
+		return validateGCPRegistry(r)
+	case types.RegistryGeneric:
+		return validateGenericRegistry(r)
+	}
+
+	return nil
+}
+
+func errorWithType(r Registry, info string) error {
+	return fmt.Errorf(`registry of type "%s" %s`, r.Type, info)
+}
+func validateAWSRegistry(r Registry) error {
+	if r.AWS.Region == "" {
+		return errorWithType(r, "requires a field region")
 	}
+	if r.AWS.AccountID == "" {
+		return errorWithType(r, `requires a field "accountdId"`)
+	}
+	return nil
+}
+
+func validateGCPRegistry(r Registry) error {
+	if r.GCP.Location == "" {
+		return errorWithType(r, `requires a field "location"`)
+	}
+	if r.GCP.ProjectID == "" {
+		return errorWithType(r, `requires a field "projectId"`)
+	}
+	if r.GCP.RepositoryID == "" {
+		return errorWithType(r, `requires a field "repositoryId"`)
+	}
+	return nil
+}
 
+func validateGenericRegistry(r Registry) error {
```

Comment:
Don't you need to check for `username` and `password` as well? Otherwise what would be the reason to specify this in the config?

Reply:
The username and password fields are both optional since some internal registries might not require authentication.

Reply:
I'm probably missing something. Why would you need to list the registry if it doesn't require authentication since the default behaviour is pulling without auth. 🤔

Reply:
The validate is used for source and target registries. There's no real point in defining a source registry without auth, but for a target registry, you might have something internal that doesn't need creds, for example an in-cluster registry.

For example, I use this config:

```yaml
source:
    filters:
    - jmespath: obj.metadata.namespace == 'k8s-image-swapper'
    registries:
      - type: generic
        generic:
          username: ${dockerhub_user}
          password: ${dockerhub_token}
          repository: docker.io
      - type: generic
        generic:
          username: ${quay_user}
          password: ${quay_token}
          repository: quay.io
  target:
    type: generic
    generic:
      repository: docker-private-registry.registry.svc.cluster.local
```